src/contracts/core/AVSDirectory.sol

```diff
@@ -490,4 +491,381 @@ contract AVSDirectory is
+            bytes32 opSetKey = _hashStrategyAndOperatorSet(strategies[i], opSet);
+            address[] memory allocators = _operatorAllocators[operator][opSetKey];
+
+            // For each strategy, slash each of its allocators and reduce their total magnitudes
+            for (uint256 j = 0; j < allocators.length; ++j) {
+                _slashAllocator(operator, opSetKey, allocators[j], bipsToSlash);
+                emit OperatorSlashed(operator, strategies[i], opSet, allocators[j], bipsToSlash);
+            }
+        }
+    }
+
+    /**
+     *
+     *                         INTERNAL FUNCTIONS
+     *
+     */
+
+    function _queueAllocation(
+        address allocator,
+        address operator,
+        IStrategy strategy,
+        MagnitudeAdjustment[] calldata magnitudeAdjustments,
+        AllocatorOperatorDetails storage allocatorDetails
+    ) internal returns (QueuedReallocation[] memory) {
+        TotalAndNonslashableUpdate storage totalMagnitudeUpdate = _getLatestTotalAndNonslashableUpdate(operator, strategy, allocator);
+        uint64 nonslashableMagnitude = totalMagnitudeUpdate.nonslashableMagnitude;
+        uint32 nonce = allocatorDetails.reallocationNonce;
+
+        QueuedReallocation[] memory queuedAllocations = new QueuedReallocation[](magnitudeAdjustments.length);
+
+        for (uint256 i = 0; i < magnitudeAdjustments.length; ++i) {
+            uint64 scalingFactor = _getCurrentScalingFactor(
+                operator,
+                _hashStrategyAndOperatorSet(strategy, magnitudeAdjustments[i].operatorSet)
+            );
+            uint64 normalizedMagnitudeIncrement = SlashingAccountingUtils.normalizeMagnitude({
+                nonNormalizedMagnitude: magnitudeAdjustments[i].magnitudeDiff,
+                scalingFactor: scalingFactor
+            });
+            require(
+                normalizedMagnitudeIncrement <= nonslashableMagnitude,
+                "AVSDirectory._queueAllocation: normalized magnitude increment exceeds nonslashable magnitude"
+            );
+            
+            queuedAllocations[i] = QueuedReallocation({
+                magnitudeAdjustmentType: MagnitudeAdjustmentType.ALLOCATION,
+                allocator: allocator,
+                operator: operator,
+                operatorSet: magnitudeAdjustments[i].operatorSet,
+                strategy: strategy,
+                magnitudeDiff: magnitudeAdjustments[i].magnitudeDiff,
+                nonce: nonce,
+                queuedScalingFactor: scalingFactor,
+                timestamp: uint32(block.timestamp)
+            });
+
+            // set queued allocation hash struct in mapping as pending
+            bytes32 reallocationHash = calculateQueuedReallocationHash(queuedAllocations[i]);
+            queuedReallocationHashes[reallocationHash] = true;
+
+            // decrement from non slashable magnitude
+            nonslashableMagnitude -= normalizedMagnitudeIncrement;
```

Comment:
need to overwrite storage

Reply:
are we doing unbacked increases? Or backed increases.
If the latter then we need the same running queued allocation sum as before

Reply:
yep we should do the latter imo.

however, it kinda sucks that the allocations take two steps, since the UI for stakers is "possibly more risky" wonder if we could get rid of that